### PR TITLE
Improve: use another field type as primary key

### PR DESCRIPTION
**Pull request by lucianobapo**
Now the uuid can be done

### CODE REVIEW

**Comment by tabacitu** [comment]
Hi @lucianobapo . Thanks for the PR. Can you please explain what the problem is - what this PR fixes?

Thanks a lot!

**Comment by lucianobapo** [comment]
Hello. With this change we have choice to change the primary key default.
I done this change because i use uuid fied as a key to edit records for security reasons.

File: src/PanelTraits/Read.php [hunk 1]
```diff
@@ -20,7 +20,9 @@
     public function getEntry($id)
     {
         if (! $this->entry) {
-            $this->entry = $this->model->findOrFail($id);
+            if (is_null($this->entry = $this->model->where($this->model->getKeyName(), $id)->get()->first())) {
```
Comment (lloy0076) [inline src/PanelTraits/Read.php]:
```
$this->entry = $this->model ... ->firstOrFail();
```

**Comment by lloy0076** [comment]
I can't recall correctly but shouldn't `findOrFail` use `getKeyName` or does it blindly just check `id`?

**Comment by lucianobapo** [comment]
Actually, after testing, "findOrFail($id)" will search on {'*'] columns by default. It means if we change the KeyName "getKeyName()" then it will works also. 

I changed the keyName from "id" to "uuid" fields and works. I override function bootIfNotBooted() from Model and put inside it "$this->setKeyName('uuid');"

This PR is not necessarry anymore. Sorry for this, lets close it

**Comment by tabacitu** [comment]
@lucianobapo no problem, thanks for getting to the bottom of it and sharing your results. Cheers!

**Pull request closed without merging**